test/unit/services/dataAccessLayer.test.js

```diff
@@ -0,0 +1,62 @@
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+
+const userQuery = require("../../../models/users");
+const sinon = require("sinon");
+const { retrieveUsers, removeSensitiveInfo } = require("../../../services/dataAccessLayer");
+const userData = require("../../fixtures/user/user")();
+const { sensitiveData } = require("../../../constants/users");
+chai.use(chaiHttp);
+const expect = chai.expect;
+
+describe("Data Access Layer", function () {
+  describe("retrieveUsers", function () {
+    it("should fetch a single user by ID and remove sensitive info", async function () {
+      const fetchUserStub = sinon.stub(userQuery, "fetchUser");
+      fetchUserStub.returns(Promise.resolve({ user: userData[12] }));
+
+      const result = await retrieveUsers({ id: userData[12].id });
+      removeSensitiveInfo(userData[12]);
+      expect(result.user).to.deep.equal(userData[12]);
+      sensitiveData.forEach((key) => {
+        expect(userData[12]).to.not.have.property(key);
+      });
+    });
+
+    it("should fetch users by username and remove sensitive info", async function () {
+      const fetchUserStub = sinon.stub(userQuery, "fetchUsers");
+      fetchUserStub.returns(Promise.resolve({ users: [userData[12]] }));
```

Comment:
if we try to fetch by user id, won't it be good enough because id's are always unique or username should be made unique and use it